File: Algorithm/0x3f/滑动窗口/不定长滑窗/越长越合法(求最短最小)/长度最短的子数组.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0, right, n = nums.length;
        int res = n + 1, window = 0;

        for(right = 0; right < n; right++){
            window += nums[right];
            while(window >= target){
                res = Math.min(res, right - left + 1);
                window -= nums[left];
                left++;
            }
        }
        return res == n + 1 ? 0 : res;
    }
}

File: Algorithm/0x3f/滑动窗口/不定长滑窗/越长越合法(求最短最小)/长度最短的子数组.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int res = 1e9, window = 0, n = nums.size();
        int left = 0, right = 0;
        for(right = 0; right < n; right++)
        {
            window += nums[right];
            while(window >= target)
            {
                res = min(res, right - left + 1);
                window -= nums[left];
                left++;
            }
        }

        return res == 1e9 ? 0 : res;
    }
};
